deployment/pytorch/torch-hub-yolov5.ipynb

```diff
@@ -0,0 +1,910 @@
+        "        \"opencv-python-headless == 4.6.0.66\",\n",
+        "        \"psutil == 5.9.4\",\n",
+        "        \"seaborn == 0.12.1\",\n",
+        "        \"torch == 1.13.0\",\n",
+        "        \"torchvision == 0.14.0\",\n",
+        "        \"transformers == 4.25.1\",\n",
+        "        \"verta == 0.21.1\",\n",
+        "    ])\n",
+        ")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "60NCs0j8KHvg"
+      },
+      "source": [
+        "## 7. Auto Scaling and Resources"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "LBLEtVauKBNP"
+      },
+      "outputs": [],
+      "source": [
+        "autoscaling = Autoscaling(min_replicas=1, max_replicas=20, min_scale=0.1, max_scale=10)\n",
+        "autoscaling.add_metric(CpuUtilizationTarget(0.6))\n",
+        "autoscaling.add_metric(MemoryUtilizationTarget(0.7))\n",
+        "autoscaling.add_metric(RequestsPerWorkerTarget(1))\n",
+        "resources = Resources(cpu=2., memory='12Gi')     "
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Y0gm6fOse6BP"
+      },
+      "source": [
+        "## 8. Model Endpoint"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "wZi8X-LoCORa"
+      },
+      "outputs": [],
+      "source": [
+        "endpoint = client.get_or_create_endpoint(NAME)"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "F4-9M2_xKF7R"
+      },
+      "outputs": [],
+      "source": [
+        "torch_cache_env = {'TORCH_HOME': '/tmp'}"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "OHLC3QefHeFw",
+        "outputId": "c8af5b4d-621b-4cd3-f8bb-aa46a9d9b653"
+      },
+      "outputs": [],
+      "source": [
+        "%%time\n",
+        "endpoint.update(\n",
+        "    model,\n",
+        "    autoscaling = autoscaling,\n",
+        "    resources = resources,\n",
+        "    env_vars = torch_cache_env,\n",
+        "    wait = True\n",
+        ")"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "p7x7dfWoCNxK"
+      },
+      "outputs": [],
+      "source": [
+        "deployed_model = endpoint.get_deployed_model()"
+      ]
+    },
+    {
+      "attachments": {},
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "## 9. Predictions"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "4nS6rzzyK58c"
+      },
+      "outputs": [],
+      "source": [
+        "def process_image(url):\n",
+        "    return deployed_model.predict(url)"
```

Comment:
Because you changed

```diff
          "urls = [\n",
-         "    ['http://www.cs.ucf.edu/~aroshan/index_files/Dataset_PitOrlManh/images/000001_0.jpg'],\n",
-         "    ['http://www.cs.ucf.edu/~aroshan/index_files/Dataset_PitOrlManh/images/000001_1.jpg'],\n",
-         "    ['http://www.cs.ucf.edu/~aroshan/index_files/Dataset_PitOrlManh/images/000001_2.jpg'],\n",
-         "    ['http://www.cs.ucf.edu/~aroshan/index_files/Dataset_PitOrlManh/images/000001_3.jpg'],\n",
-         "    ['http://www.cs.ucf.edu/~aroshan/index_files/Dataset_PitOrlManh/images/000001_4.jpg'],\n",
-         "    ['http://www.cs.ucf.edu/~aroshan/index_files/Dataset_PitOrlManh/images/000001_5.jpg']\n",
+         "    'http://www.cs.ucf.edu/~aroshan/index_files/Dataset_PitOrlManh/images/000001_0.jpg',\n",
+         "    'http://www.cs.ucf.edu/~aroshan/index_files/Dataset_PitOrlManh/images/000001_1.jpg',\n",
+         "    'http://www.cs.ucf.edu/~aroshan/index_files/Dataset_PitOrlManh/images/000001_2.jpg',\n",
+         "    'http://www.cs.ucf.edu/~aroshan/index_files/Dataset_PitOrlManh/images/000001_3.jpg',\n",
+         "    'http://www.cs.ucf.edu/~aroshan/index_files/Dataset_PitOrlManh/images/000001_4.jpg',\n",
+         "    'http://www.cs.ucf.edu/~aroshan/index_files/Dataset_PitOrlManh/images/000001_5.jpg'\n",
          "]"
```

in your last commit (ff559f2c96bb20a93623f90884889209c69402da), we are now passing `"url"` instead of `["url"]`, which violates the model API (`Predictor.example()` returns `["url1", "url2"]`). It still works because `ultralytics/yolov5` is clever, but we should try to be consistent.

We can just wrap this in a list

```suggestion
        "    return deployed_model.predict([url])"
```

Reply:
Perfect. The ModelAPI input and the actual model input is something that can easily be overlooked :fearful: